js/homework_OOP.js

```diff
@@ -0,0 +1,82 @@
+
+class BasicUnit {
+  constructor(speed, attack, health, phrase) {
+    this.speed = speed;
+    this.attack = attack;
+    this.health = health;
+    this.phrase = phrase;
+  }
+
+  damage = () => console.log(`My damage ${this.attack}`);
+  move = () => console.log(`I'm moving at the speed of ${this.speed}!`);
+}
+
+class Cood extends BasicUnit {
+  constructor(speed, attack, health, phrase, aura) {
+    super(speed, attack, health, phrase);
+    this.aura = aura;
+  }
+  crit = () =>console.log(`With aura of ${this.aura} you have a 10% chance to deal double damage` );
+}
+
+class Evil extends BasicUnit {
+  constructor(speed, attack, health, phrase, armor) {
+    super(speed, attack, health, phrase);
+    this.armor = armor;
+  }
+  defense = () => console.log(`Having additional protection ${this.armor}, damage to you is reduced by 5%`)
+}
+
+class Human extends Cood {
+	#amuletFaith
+	constructor(speed, attack, health, phrase, aura, sword, amuletFaith) {
+		super(speed, attack, health, phrase, aura)
+		this.sword = sword
+		this.amuletFaith = amuletFaith
```

Comment:
Для того, щоб властивість була приватною:
```suggestion
		this.#amuletFaith = amuletFaith
```

Спробуйте чи доступний метод ззовні у себе